packages/flutter/lib/src/widgets/animated_child_switcher.dart

```diff
@@ -53,17 +148,69 @@ class AnimatedChildSwitcher extends StatefulWidget {
   /// [duration].
   final Widget child;
 
+  /// The duration over which to perform the cross fade using [FadeTransition].
+  final Duration duration;
+
   /// The animation curve to use when fading in the current widget.
   final Curve switchInCurve;
 
   /// The animation curve to use when fading out the previous widgets.
   final Curve switchOutCurve;
 
-  /// The duration over which to perform the cross fade using [FadeTransition].
-  final Duration duration;
+  /// The builder function used to generate custom transitions for
+  /// [AnimatedChildSwitcher].
+  ///
+  /// The default is [AnimatedChildSwitcher.defaultTransitionBuilder].
+  ///
+  /// See also:
+  ///
+  ///  * [AnimatedChildSwitcherTransitionBuilder] for more information about
+  ///    how a transition builder should function.
+  final AnimatedChildSwitcherTransitionBuilder transitionBuilder;
+
+  /// The builder function used to generate custom child layouts for
+  /// [AnimatedChildSwitcher].
+  ///
+  /// The default is [AnimatedChildSwitcher.defaultLayoutBuilder].
```

Comment:
It would be more helpful to say what this builder function is for:  A function that wraps all of the children that are in the midst of transitioning out and the [child] that's transitioning in, with a widget that lays all of them out.

Reply:
Done.